File: app/src/main/java/ru/org/adons/cblock/ui/activity/IMainListener.java
package ru.org.adons.cblock.ui.activity;

import ru.org.adons.cblock.ui.fragment.IBaseFragmentListener;
import ru.org.adons.cblock.ui.view.add.AddFragment;
import ru.org.adons.cblock.ui.view.main.MainFragment;

/**
 * Listen {@link MainFragment}, {@link AddFragment} and notify {@link MainActivity}
 */
public interface IMainListener extends IBaseFragmentListener {

    void showAddFragment();

    void onBackPressed();

}

File: app/src/main/java/ru/org/adons/cblock/ui/activity/IMainListener.kt
package ru.org.adons.cblock.ui.activity

import ru.org.adons.cblock.ui.view.add.AddFragment
import ru.org.adons.cblock.ui.view.main.MainFragment

/**
 * Listen [MainFragment], [AddFragment] and notify [MainActivity]
 */
interface IMainListener {

    val mainComponent: MainComponent

    fun showAddFragment()

    fun showProgress()

    fun hideProgress()

    fun showError(message: String)

    fun onBackPressed()

}
